src/chocolatey/infrastructure.app/nuget/NugetCommon.cs

```diff
@@ -451,16 +451,27 @@ ISet<SourcePackageDependencyInfo> dependencyInfos
 
             //if (availablePackages.Contains(packageID)) return;
 
-            var dependencyInfoResources = resources.DependencyInfoResources();
-
-            foreach (var dependencyInfoResource in dependencyInfoResources)
+            foreach (var resource in resources)
             {
-                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = Array.Empty<SourcePackageDependencyInfo>();
+                var dependencyInfoResource = resource.DependencyInfoResource;
+
+                if (dependencyInfoResource is null)
+                {
+                    // We can't lookup any dependencies using this resource.
+                    continue;
+                }
+
+                // check if we already have packages matching the constraints in the list
+                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = availablePackages.Where(
+                    p => string.Equals(p.Id, packageId, StringComparison.OrdinalIgnoreCase) && p.HasVersion && versionRange.Satisfies(p.Version) && p.Source == resource.Source).ToList();

```

Comment:
From what I can see looking over things here, the list that is generated here (if there are any available packages that match) is only ever checked with `.Any()` on line 470, then again on line 485, before being converted back to a list on line 493. So my question is: Do we need to use `.ToList()` here, or can it be left off?